### PR TITLE
KREST-561: Add retries for flaky tests.

### PR DESCRIPTION
**Pull request by rigelbm**
All of the tests below are flaky due to lack of consistency for admin operations in Kafka.

Test failures addressed:
* https://confluentinc.atlassian.net/browse/KREST-522
* https://confluentinc.atlassian.net/browse/KREST-527
* https://confluentinc.atlassian.net/browse/KREST-542
* https://confluentinc.atlassian.net/browse/KREST-546
* https://confluentinc.atlassian.net/browse/KREST-547
* https://confluentinc.atlassian.net/browse/KREST-555

### CODE REVIEW

File: kafka-rest/src/test/java/io/confluent/kafkarest/TestUtils.java [hunk 1]
```diff
@@ -246,33 +246,29 @@ public static void assertPartitionOffsetsEqual(List<PartitionOffset> a, List<Par
...
-      Duration initialBackoff) {
+  public static void testWithRetry(Runnable assertions, int numRetries, Duration initialBackoff) {
     Duration backoff = initialBackoff;
-    for (int i = 0; i <= numRetries; i++) {
-      if (testCondition.get()) {
+    for (int i = 0;; i++) {
+      try {
+        assertions.run();
         return;
+      } catch (Throwable e) {
```
Comment (dimitarndimitrov) [inline kafka-rest/src/test/java/io/confluent/kafkarest/TestUtils.java:259]: super-nit: The typical naming would be `Throwable t`.

File: kafka-rest/src/test/java/io/confluent/kafkarest/integration/v3/BrokerConfigsResourceIntegrationTest.java [hunk 2]
```diff
@@ -358,18 +359,21 @@ public void updateAndReset_existingConfig_returnsDefaultUpdatedAndDefaultAgain()
...
-            .accept(MediaType.APPLICATION_JSON)
-            .get();
-    assertEquals(Status.OK.getStatusCode(), responseAfterUpdate.getStatus());
-
-    GetBrokerConfigResponse actualAfterUpdate =
-        responseAfterUpdate.readEntity(GetBrokerConfigResponse.class);
-    assertEquals(expectedAfterUpdate, actualAfterUpdate);
+    testWithRetry(
+        () -> {
+          Response responseAfterUpdate =
```
Comment (dimitarndimitrov) [inline kafka-rest/src/test/java/io/confluent/kafkarest/integration/v3/BrokerConfigsResourceIntegrationTest.java:364]: I realise it's according to the style guide, but that indentation looks truly awful.

**Review by dimitarndimitrov — APPROVED** [review]
+1.

File: kafka-rest/src/test/java/io/confluent/kafkarest/integration/v3/BrokerConfigsResourceIntegrationTest.java (hunk 2 above)
Comment (rigelbm) [reply kafka-rest/src/test/java/io/confluent/kafkarest/integration/v3/BrokerConfigsResourceIntegrationTest.java:364]: Done.

File: kafka-rest/src/test/java/io/confluent/kafkarest/TestUtils.java (hunk 1 above)
Comment (rigelbm) [reply kafka-rest/src/test/java/io/confluent/kafkarest/TestUtils.java]: Done.

**Pull request merged**